elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -96,9 +96,18 @@ def resurrect_dead_connections!
         def __rebuild_connections(arguments={})
           @hosts       = arguments[:hosts]    || []
           @options     = arguments[:options]  || {}
+          __close_connections # close old connections
           @connections = __build_connections
         end
 
+        # Closes the connections collection.
+        #
+        # @api private
+        #
+        def __close_connections
+          @connections.each {|c| c.dead! }
```

Comment:
The idea is that after calling __close_connections no more requests should be performed which will work since perform_request fetches a "non-dead" connection every time, iirc